ml_garden/core/steps/encode.py

```diff
@@ -277,70 +277,76 @@ def _get_encoder_class_and_params(
 
         return encoder, encoder_params
 
+    def _log_encoder_override(
+        self, feature, encoder_class, high_cardinality_features, low_cardinality_features
+    ):
+        if feature in high_cardinality_features:
+            self.logger.info(
+                f"Feature '{feature}' encoder overridden from"
+                f" {self.HIGH_CARDINALITY_ENCODER} to {encoder_class.__name__}"
+            )
+        elif feature in low_cardinality_features:
+            self.logger.info(
+                f"Feature '{feature}' encoder overridden from {self.LOW_CARDINALITY_ENCODER} to"
+                f" {encoder_class.__name__}"
+            )
+        else:
+            self.logger.info(
+                f"Feature '{feature}' explicitly encoded with {encoder_class.__name__}"
+            )
+
     def _create_column_transformer(
         self,
         high_cardinality_features: List[str],
         low_cardinality_features: List[str],
         numeric_features: List[str],
     ) -> ColumnTransformer:
-        """Create a ColumnTransformer for encoding.
-
-        If `feature_encoders` is not provided, use the default encoders based on cardinality.
-        * For low cardinality features, use OrdinalEncoder.
-        * For high cardinality features, use TargetEncoder.
-        * For numeric features, pass them as is.
-
-        Parameters
-        ----------
-        high_cardinality_features : List[str]
-            High cardinality features
-        low_cardinality_features : List[str]
-            Low cardinality features
-        numeric_features : List[str]
-            Numeric features
-        Returns
-        -------
-        ColumnTransformer
-            The ColumnTransformer
-        """
         transformers = []
-
-        if not self.feature_encoders:
-            high_cardinality_encoder_class, high_cardinality_encoder_params = (
-                self._get_encoder_class_and_params(self.HIGH_CARDINALITY_ENCODER)
-            )
-            low_cardinality_encoder_class, low_cardinality_encoder_params = (
-                self._get_encoder_class_and_params(self.LOW_CARDINALITY_ENCODER)
+        encoded_features = set()
+
+        # Handle shared encoder for Origin and Destination
+        if "shared_origin_destination" in self.feature_encoders:
```

Comment:
Hmmm, this feels a bit too specific. Like origin and destination are the columns in this particular dataset right?
Do we even need to share encoding between the columns?
Either if we use OrdinalEncoding or TargetEncoding, the tree will learn the cuts it needs to do for each encode separately and there would be no difference right? Only the cut thresholds but the results should be the same?